src/main/kotlin/io/github/yuutoproject/yuutobot/commands/Dialog.kt

```diff
@@ -20,45 +20,144 @@ package io.github.yuutoproject.yuutobot.commands
+        "bath",
+        "beach",
+        "cabin",
+        "camp",
+        "cave",
+        "forest",
+        "messhall"
+    )
+
+    private val characters: List<String> = listOf(
+        "aiden",
+        "avan",
+        "chiaki",
+        "connor",
+        "eduard",
+        "felix",
+        "goro",
+        "hiro",
+        "hunter",
+        "jirou",
+        "keitaro",
+        "kieran",
+        "knox",
+        "lee",
+        "naoto",
+        "natsumi",
+        "seto",
+        "taiga",
+        "yoichi",
+        "yoshi",
+        "yuri",
+        "yuuto"
+    )
+
+    private val backgroundsString: String = "`${backgrounds.joinToString("`, `")}`"
+    private val charactersString: String = "`${characters.joinToString("`, `")}`"
+
+    // Using ExperimentalStdLib for .removeFirst() in Mutable Lists
+    @ExperimentalStdlibApi
     override fun run(args: MutableList<String>, event: GuildMessageReceivedEvent) {
-        val moveThisToAStorageLocation = OkHttpClient()
+        val now: Long = System.currentTimeMillis()
+
+        if (args.count() < 2) {
+            event.channel.sendMessage("This command requires two arguments : `dialog [background] <character> <text>` ([] is optional)").queue()
+            return
+        }
+
+        var character: String = args.removeFirst().toLowerCase()
+        val background: String
+
+        if (characters.contains(character)) {
+            background = "camp"
+        } else {
+            background = character
+            character = args.removeFirst().toLowerCase()
+        }
+
+        if (!backgrounds.contains(background)) {
+            event.channel.sendMessage("Sorry but I couldn't find `$background` as a location\nAvailable backgrounds are: $backgroundsString").queue()
+            return
+        }
+
+        if (!characters.contains(character)) {
+            event.channel.sendMessage("Sorry, but I don't think that `$character` is a character in Camp Buddy\nAvailable characters are: $charactersString").queue()
+            return
+        }
+
+        if (args.count() < 1) {
+            event.channel.sendMessage("Please provide a message to be written~!").queue()
+            return
+        }
+
+        val text: String = args.joinToString(" ").replace("/[‘’]/g".toRegex(), "'")
+
+        if (text.length > 120) {
+            event.channel.sendMessage("Sorry, the message limit is 120 characters <:hiroJey:692008426842226708>").queue()
+            return
+        }
+
+        if (
+            EMOTE_MENTIONS_REGEX.containsMatchIn(text) ||
+            EMOJI_REGEX.containsMatchIn(text) ||
+            NONASCII_REGEX.containsMatchIn(text)
+        ) {
+            event.channel.sendMessage("Sorry, I can't display emotes, mentions, or non-latin characters").queue()
+            return
+        }
 
         event.channel.sendTyping().queue()
 
         val json = "application/json; charset=utf-8".toMediaType()
         val body = """{
```

Comment:
A better way would be
```java
DataObject.empty()
    .put("background", background)
    .put("character", character)
    .put("text", text)
    .toString().toRequestBody(json)
```

Reply:
Okay ill try that, i didn't knew the proper way of handling JSON's on kotlin so i let it be

Reply:
It's not a kotlin thing but a class from JDA, it uses fasterxml's Jackson internally so it has quote protection etc